.github/workflows/main.yml

```diff
@@ -1,28 +1,36 @@
-name: Run pyright
-on:
-  push:
-    branches:
-      - main  # Runs on push to main branch
-  pull_request:
-    branches:
-      - main  # Runs on PRs targeting main branch
-jobs:
-  type-check:
-    runs-on: ubuntu-latest
-    steps:
-      - name: checkout  repository
-        uses: actions/checkout@v4
-
-      - name: Configure Py env
-        run: |
-           python -m venv .venv
-           . .venv/bin/activate
-           pip install -r requirements.txt
-           
-      - name: Add virtual environ to PATH
-        run: echo "$PWD/.venv/bin" >> $GITHUB_PATH
-
-      - name: Run Pyright 
-        uses: jakebailey/pyright-action@v2.3.2
-        with:
-           warnings: true
+name: Main workflow
+on:
+  push:
+
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest

```

Comment:
Looks like we might need to change this to "ubuntu-20.04" for Python 3.8.10 to work.